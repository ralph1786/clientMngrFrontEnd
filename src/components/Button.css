.button {
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  margin: 2%;
  text-align: center;
}

.button a:nth-child(odd) {
  grid-column: 4/6;
  width: 100px;
}

.button a:nth-child(even) {
  grid-column: 8/10;
  width: 100px;
}

.btn-hover {
  width: 200px;
  font-size: 1.6em;
  font-weight: 600;
  color: #fff;
  cursor: pointer;
  margin: 20px;
  height: 55px;
  text-align: center;
  border: none;
  background-size: 300% 100%;

  border-radius: 50px;
  moz-transition: all 0.4s ease-in-out;
  -o-transition: all 0.4s ease-in-out;
  -webkit-transition: all 0.4s ease-in-out;
  transition: all 0.4s ease-in-out;
}

.btn-hover:hover {
  background-position: 100% 0;
  moz-transition: all 0.4s ease-in-out;
  -o-transition: all 0.4s ease-in-out;
  -webkit-transition: all 0.4s ease-in-out;
  transition: all 0.4s ease-in-out;
}

.btn-hover:focus {
  outline: none;
}

.btn-hover.color-1 {
  background-image: linear-gradient(
    to right,
    #25aae1,
    #40e495,
    #30dd8a,
    #2bb673
  );
  box-shadow: 0 4px 15px 0 rgba(49, 196, 190, 0.75);
}
