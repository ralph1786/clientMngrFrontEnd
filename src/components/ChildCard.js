import React from "react";
import { connect } from "react-redux";
import { deleteChild } from "../actions/childActions";
import { Link, withRouter } from "react-router-dom";
import "./ChildCard.scss";

const ChildCard = props => {
  // console.log(props.child);
  const { name, age, image, address, allergies, balance } = props.child;
  return (
    <div className="child-card animated fadeInUp">
      <h3>Name: {name}</h3>
      <p>Age: {age}</p>
      <img src={image} alt="children" />
      <address>Address: {address}</address>
      <p>Allergies: {allergies}</p>
      <p>Balance: ${balance}</p>
      <Link to="/edit">
        <button>Edit</button>
      </Link>
      <button onClick={() => props.deleteChild(props.child.id)}>Delete</button>
    </div>
  );
};

const mapDispatchToProps = dispatch => {
  return {
    deleteChild: id => dispatch(deleteChild(id))
  };
};

export default withRouter(
  connect(
    null,
    mapDispatchToProps
  )(ChildCard)
);
