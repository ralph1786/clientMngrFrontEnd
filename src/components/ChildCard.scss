.child-card {
  -webkit-box-shadow: 10px 10px 29px 1px rgba(140, 137, 140, 1);
  -moz-box-shadow: 10px 10px 29px 1px rgba(140, 137, 140, 1);
  box-shadow: 10px 10px 29px 1px rgba(140, 137, 140, 1);
  margin: 10px;
  padding: 5px;
  width: 400px;
  height: 550px;
  overflow: hidden;
  text-align: center;
}

.child-card img {
  width: 300px;
  height: 300px;
  border-radius: 10px;
  margin-bottom: 10px;
}

.child-card button:nth-child(odd) {
  margin: 10px;
  padding: 2px 20px;
  border-radius: 25px;
}

.child-card button:nth-child(odd):hover {
  background-color: orange;
  color: white;
  cursor: pointer;
}

.child-card button:nth-child(even) {
  margin: 10px;
  padding: 2px 20px;
  border-radius: 25px;
}

.child-card button:nth-child(even):hover {
  background-color: rgb(248, 44, 44);
  color: white;
  cursor: pointer;
}
