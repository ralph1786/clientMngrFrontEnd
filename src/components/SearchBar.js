import React from "react";

const SearchBar = props => {
  return (
    <div>
      <input
        type="text"
        value={props.searchWord}
        onChange={props.onChange}
        placeholder="Search Child"
      />
    </div>
  );
};

export default SearchBar;
