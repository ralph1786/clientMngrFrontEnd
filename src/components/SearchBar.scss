.search-input {
  position: relative;
  border-radius: 25px;
}

.search-input-icon {
  position: absolute;
  top: 11%;
  right: 36.25%;
  font-size: 20px;
}

@media (max-width: 768px) {
  .search-input-icon {
    top: 8.25%;
    font-size: 15px;
  }
}

@media (max-width: 414px) {
  .search-input-icon {
    top: 10.75%;
    right: 25%;
    font-size: 12px;
  }
}

@media (max-width: 320px) {
  .search-input-icon {
    top: 11%;
    font-size: 10px;
  }
}
