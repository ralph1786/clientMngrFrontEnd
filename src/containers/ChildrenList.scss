@import "../utilities/_variables";

.children-list {
  display: grid;
  grid-template-columns: repeat(auto-fit, minmax(100px, 400px));
  grid-template-rows: repeat(3, 1fr);
  margin-left: 12%;
}

.search-bar input {
  margin-left: 34%;
  width: 30%;
  height: 30px;
  font-size: 1.3em;
  margin-bottom: 7%;
  margin-top: 6%;
  outline: none;
  text-align: center;
  border: 1px solid rgb(68, 67, 67);

  &:focus {
    -webkit-box-shadow: 6px 6px 24px 1px $button-color-hover-opacity;
    -moz-box-shadow: 6px 6px 24px 1px $button-color-hover-opacity;
    box-shadow: 6px 6px 24px 1px $button-color-hover-opacity;
  }
}

@media (max-width: 768px) {
  .children-list {
    margin-left: 25%;
  }
  .search-bar input {
    height: 25px;
    margin-top: 10.5%;
  }
}

@media (max-width: 414px) {
  .children-list {
    margin-left: 15%;
  }
  .search-bar input {
    width: 50%;
    height: 18px;
    margin-left: 25%;
    font-size: 1.1em;
    margin-top: 18%;
  }
}

@media (max-width: 375px) {
  .children-list {
    margin-left: 10%;
  }
}

@media (max-width: 320px) {
  .search-bar input {
    margin-left: 25%;
    height: 16px;
    font-size: 0.9em;
  }
  .children-list {
    margin-left: 4%;
  }
}
