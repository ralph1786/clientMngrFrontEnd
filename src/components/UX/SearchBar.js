import React, { Fragment } from "react";
import "./SearchBar.scss";

const SearchBar = props => {
  return (
    <Fragment>
      <input
        className="search-input"
        type="text"
        value={props.searchWord}
        onChange={props.onChange}
        placeholder="Search Child"
      />
      <span className="search-input-icon">
        <i className="fas fa-search fa-lg" />
      </span>
    </Fragment>
  );
};

export default SearchBar;
