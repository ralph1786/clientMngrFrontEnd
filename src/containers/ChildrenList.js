import React, { Component, Fragment } from "react";
import ChildCard from "../components/ChildCard";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { allChildren, searchWord } from "../actions/childActions";
import SearchBar from "../components/UX/SearchBar";
import "./ChildrenList.scss";

export class ChildrenList extends Component {
  componentDidMount() {
    this.props.allChildren();
  }

  changeHandler = e => {
    this.props.search(e.target.value);
  };

  childrenFilteredList = () => {
    const listChildren = this.props.children
      .filter(child =>
        child.name.toLowerCase().includes(this.props.searchTerm.toLowerCase())
      )
      .sort((a, b) => (a.name > b.name ? 1 : -1))
      .map(child => <ChildCard key={child.id} child={child} />);
    return listChildren;
  };

  render() {
    return (
      <Fragment>
        <div className="search-bar">
          <SearchBar
            searchWord={this.props.searchWord}
            onChange={this.changeHandler}
          />
        </div>
        <ul className="children-list">{this.childrenFilteredList()}</ul>
      </Fragment>
    );
  }
}

const mapStateToProps = state => {
  return {
    children: state.children.children,
    searchTerm: state.children.searchWord
  };
};

const mapDispatchToProps = dispatch => {
  return {
    allChildren: () => dispatch(allChildren()),
    search: word => dispatch(searchWord(word))
  };
};

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(ChildrenList)
);
