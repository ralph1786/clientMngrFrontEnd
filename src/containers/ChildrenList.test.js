import React from "react";
import { shallow } from "enzyme";
import { ChildrenList } from "./ChildrenList";
import SearchBar from "../components/UX/SearchBar";
import { ChildCard } from "../components/ChildCard";

let wrapper;

beforeEach(() => {
  wrapper = shallow(
    <ChildrenList
      children={[]}
      searchTerm={""}
      allChildren={() => {}}
      search={() => {}}
    />
  );
});

it("captures a snapshot of the ChildrenList component", () => {
  expect(wrapper).toMatchSnapshot();
});

it("contains the SearchBar component", () => {
  expect(wrapper.find(SearchBar)).toHaveLength(1);
});

it("expects to show a list of ChildCard components", () => {
  expect(wrapper.find(ChildCard)).toBeTruthy();
});
