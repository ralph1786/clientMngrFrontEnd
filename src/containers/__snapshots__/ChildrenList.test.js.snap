// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`captures a snapshot of the ChildrenList component 1`] = `
<Fragment>
  <div
    className="search-bar"
  >
    <SearchBar
      onChange={[Function]}
    />
  </div>
  <ul
    className="children-list"
  />
</Fragment>
`;
